lib/vanagon/platform/windows.rb

```diff
@@ -66,6 +66,48 @@ def generate_nuget_package(project)
         "#{@copy} $(tempdir)/#{project.name}/#{project.name}-#{@architecture}.#{nuget_package_version(project.version, project.release)}.nupkg ./output/#{target_dir}/#{nuget_package_name(project)}"]
       end
 
+      # The specific bits used to generate a windows msi package for a given project
+      # Have changed this to reflect the overall commands we need to generate the package.
+      # Question - should we break this down into some simpler Make tasks ?
+      # 1. Heat the directory tree to produce the file list
+      # 2. Compile (candle) all the wxs files into wixobj files
+      # 3. Run light to produce the final MSI
+      #
+      # @param project [Vanagon::Project] project to build a msi package of
+      # @return [Array] list of commands required to build an msi package for the given project from a tarball
+      def generate_msi_package(project)
+        target_dir = project.repo ? output_dir(project.repo) : output_dir
+        cg_name = "compfiles"
+        dir_ref = "INSTALLDIR"
+        # Actual array of commands to be written to the Makefile
+        ["mkdir -p output/#{target_dir}",
+        "mkdir -p $(tempdir)/#{project.name}/staging",
+        "gunzip -c #{project.name}-#{project.version}.tar.gz | '#{@tar}' -C '$(tempdir)/#{project.name}/staging' --strip-components 1 -xf -",
+        # Run the Heat command in a single pass - use the filter.xslt to filter out components for "special" treatment (i.e. services)
+        # Heat command documentation at: http://wixtoolset.org/documentation/manual/v3/overview/heat.html
+        #   dir <directory> - Traverse directory to find all sub-files and directories.
+        #   -v              - verbose
+        #   -ke             - Keep Empty directories
+        #   -indent <n>     - Indentation multiple (overrides default of 4)
+        #   -cg             - Component Group Name
+        #   -gg             - Generate GUIDS now
+        #   -dr             - Directory reference to root directories (cannot contains spaces e.g. -dr MyAppDirRef)
+        #   -sreg           - Suppress registry harvesting.
+        #   -var <variable> - Substitute File/@Source="SourceDir" with a preprocessor or a wix variable
+        #   -out <file>     - Output File
+        "cd $(tempdir)/pl-libffi/staging; \"$$WIX/bin/heat.exe\" dir tools -v -ke -indent 2 -cg #{cg_name} -gg -dr #{dir_ref} -sreg -var var.StageDir -out wix/#{project.name}-harvest.wxs",
```

Comment:
Darn. That sucks. That would have been a lot nicer to look at. Since we're including a link to this documentation, it might make more sense to only have specific explanations for special/confusing commands rather then copying the command definitions from the link here?

I still think you should remove referenced to filter.xslt until we actually implement that aspect of this code though.